userspace/libsinsp/filter/ast.h

```diff
@@ -117,9 +145,15 @@ struct SINSP_PUBLIC string_visitor: public expr_visitor
 */
 struct SINSP_PUBLIC expr
 {
+    expr();
```

Comment:
What are these two constructors used for? I think we should leave this interface as abstract as possible.

Due to the same reasoning, Isn't it redundant to have all three get_pos, set_pos, and m_pos? I think we should aim to either:
- Have only `m_pos`
- Have only `get_pos` and `set_pos` and leave each implementation store their class members
- Keep all three, but turn `expr` in a class and set `m_pos` as private and all the rest as public

Reply:
I'll get rid of the second constructor, I was using it but once I switched to using the static ::create() methods I didn't need it. About defining a default constructor, this is more of a style thing--some devs always want constructors defined unless the class is truly abstract, and I tend to follow that practice. But I can remove it since it doesn't do anything. (Someday defining a style guide would be useful). I did add default values to the pos_info idx/line/col though.

If the intent is that expr is an interface/fully abstract class, I can push down management of the position into the sub-classes. But if we want to keep the position as a part of the base class, I wouldn't have the derived classes access a protected member. It depends on who you ask, but many devs as well as sonarcloud (https://rules.sonarsource.com/cpp/RSPEC-3656) frown on it.

So I'd prefer the third option. I'll do that.